presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -799,6 +800,56 @@ std::unique_ptr<common::Filter> toFilter(
   VELOX_UNSUPPORTED("Unsupported filter found.");
 }
 
+template <TypeKind KIND>
+TypePtr fieldNamesToLowerCase(const TypePtr& type) {
+  return type;
```

Comment:
Why don't we convert the type name to lowercase here?

Reply:
we only need to convert field names in rows

this is the default implementation which is used for primitives (they don't have fields, and they can't contain rows)

I added a specialization for rows to convert field names, and specializations for arrays and maps to recursively handle any rows they contain